fe/fe-core/src/main/java/com/starrocks/planner/OlapTableParamGenerator.java

```diff
@@ -0,0 +1,341 @@
+package com.starrocks.planner;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Range;
+import com.starrocks.analysis.Expr;
+import com.starrocks.analysis.LiteralExpr;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.DistributionInfo;
+import com.starrocks.catalog.ExpressionRangePartitionInfo;
+import com.starrocks.catalog.ExpressionRangePartitionInfoV2;
+import com.starrocks.catalog.HashDistributionInfo;
+import com.starrocks.catalog.KeysType;
+import com.starrocks.catalog.ListPartitionInfo;
+import com.starrocks.catalog.LocalTablet;
+import com.starrocks.catalog.MaterializedIndex;
+import com.starrocks.catalog.MaterializedIndexMeta;
+import com.starrocks.catalog.OlapTable;
+import com.starrocks.catalog.Partition;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.PartitionType;
+import com.starrocks.catalog.RangePartitionInfo;
+import com.starrocks.catalog.Replica;
+import com.starrocks.catalog.Tablet;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.InternalErrorCode;
+import com.starrocks.common.Status;
+import com.starrocks.common.UserException;
+import com.starrocks.lake.LakeTablet;
+import com.starrocks.load.Load;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.system.SystemInfoService;
+import com.starrocks.thrift.TExprNode;
+import com.starrocks.thrift.TOlapTableIndexSchema;
+import com.starrocks.thrift.TOlapTableIndexTablets;
+import com.starrocks.thrift.TOlapTableLocationParam;
+import com.starrocks.thrift.TOlapTablePartition;
+import com.starrocks.thrift.TOlapTablePartitionParam;
+import com.starrocks.thrift.TOlapTableSchemaParam;
+import com.starrocks.thrift.TTabletLocation;
+import com.starrocks.warehouse.Warehouse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OlapTableParamGenerator {
```

Comment:
Can we just add some public static function interface for this function in OlapTableSink.java and avoid to refactor the code in whole file?

Reply:
Move these static func to OlapTableSink could work, but use OlapTableSink in DictQueryFunction seems little weird.

Reply:
Yes, it maybe a little weird, but olaptablesink is the entry point for all Loads. I think that from the perspective of stability, I am afraid that refactoring this part of the code will be risky. In the future, if the OlapTableParam is separated, the dict mapping code can be refactored.

Reply:
ok, I will move the code